unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+
+internal class BaseCellView : MonoBehaviour
+{
+    [SerializeField] Image thumbnailImage;
+    [SerializeField] GameObject loadingSpinner;
+    [SerializeField] Sprite errorThumbnail;
+
+    public event Action<Sprite> OnThumbnailFetched;
+
+    UnityWebRequest thumbnailRequest = null;
+    Texture2D thumbnailTexture;
+    Sprite thumbnail;
+
+    public void FetchThumbnail(string url)
+    {
+        if (thumbnail)
+        {
+            OnThumbnailFetched?.Invoke(thumbnail);
+        }
+        else if (thumbnailRequest == null)
+        {
+            thumbnailImage.sprite = null;
+
+            thumbnailRequest = UnityWebRequestTexture.GetTexture(url);
+            UnityWebRequestAsyncOperation op = thumbnailRequest.SendWebRequest();
+            op.completed += (_) =>
+            {
+                if (thumbnailRequest == null)
+                    return;
+
+                if (!thumbnailRequest.isNetworkError && !thumbnailRequest.isHttpError)
```

Comment:
We can use the `WebRequestSucceded` extension method here.

Reply:
Nice tip! thx! I'll do that